core/src/main/scala/kafka/utils/timer/Timer.scala

```diff
@@ -43,7 +81,7 @@ class Timer(taskExecutor: ExecutorService, tickMs: Long = 1, wheelSize: Int = 20
   def add(timerTask: TimerTask): Unit = {
     readLock.lock()
     try {
-      addTimerTaskEntry(new TimerTaskEntry(timerTask))
+      addTimerTaskEntry(new TimerTaskEntry(timerTask, timerTask.delayMs + System.currentTimeMillis()))
```

Comment:
This is a side-effecting function (it's the third point in the definition below):

"In computer science, a function or expression is said to have a side effect if it modifies some state or has an observable interaction with calling functions or the outside world"